impeller/entity/shaders/blending/porter_duff_blend.frag

```diff
@@ -36,13 +36,22 @@ f16vec4 Sample(f16sampler2D texture_sampler, vec2 texture_coords) {
   return IPHalfSampleDecal(texture_sampler, texture_coords);
 }
 
+f16vec4 IPHalfClamp(f16vec4 color) {
+  float16_t min = 0.0hf;
+  float16_t max = 1.0hf;
+  return clamp(color, min, max);
+}
+
 void main() {
   f16vec4 dst =
       texture(texture_sampler_dst, v_texture_coords) * frag_info.input_alpha;
   f16vec4 src = v_color;
-  frag_color =
+  // This currently needs a clamp so that floating point textures blend
+  // correctly in wide gamut. Remove if we switch to a fixed point extended
+  // range format.
+  frag_color = IPHalfClamp(
```

Comment:
I not certain we have to do this.  I'd rather be more pragmatic with it since it is very complicated.  We have wide gamut tests now so we can demonstrate when/where this goes wrong.  If we get a case that is broken we could use push constants or uniforms to specify the clamping range.  We'd need that for the fixed point format as well.

At a minimum it should be in its own PR.

Reply:
Removed

Reply:
If we don't clamp the alpha, then this change re-breaks the vertices, so we need it otherwise this change is unsafe.

Reply:
Yea, clamping the alpha sounds good.  The jury is still out wether clamping the color components is the correct behavior, I'm looking into that.  Can you add a TODO linking that BGRA10_XR issue above the clamp.

I'm not sure where this shader is used fully, but clamping the color components may well break our wide gamut devicelab tests.

Reply:
I'm just going to clamp the alpha, draw atlas is rendering wrong still at ToT.

https://github.com/flutter/engine/pull/51792